src/vs/platform/quickinput/common/quickInput.ts

```diff
@@ -94,4 +95,10 @@ export interface IQuickInputService {
 	 * Cancels quick input and closes it.
 	 */
 	cancel(): Promise<void>;
+
+	/**
+	 * Formats quick pick items to be pinnable
+	 * with a sticky pinned category at the top
+	 */
+	formatPinnedItems(storageKey: string, quickPick: IQuickPick<IQuickPickItem>, storageService: IStorageService, callback?: () => Promise<void>): void;
```

Comment:
A `ServicesAccessor` would probably be nicer to deal with than the storage service, so it gets called via:

```ts
instantiationService.invokeFunction(formatPinnedItems, key, quickPick);
```

That way only the general instantiation service is needed in the calling file.